### PR TITLE
Email questions

### PR DESCRIPTION
**Pull request by DmytroDidukh**
-email questions list page
-email questions details page
-pagination
-question filtering
-redux
-answer to question functionality

### CODE REVIEW

File: src/components/nav-menu/nav-menu.js [hunk 1]
```diff
@@ -50,6 +55,9 @@ const NavMenu = ({ width }) => {
           <PathIcon />
         </ListItemIcon>
         <ListItemText primary={pathTitle} />
+        {pathTitle === 'Запитання & Відповіді' && (
```
Comment (TorskyM) [inline src/components/nav-menu/nav-menu.js:58]: Please, rename this title to `Запитання та відповіді`, also move it to config

File: src/configs/menu-categories.js [hunk 2]
```diff
@@ -18,6 +19,7 @@ const menuCategories = [
   ['Контакти', routes.pathToContacts, ImportLocationOnIcon],
   ['Гобелени', routes.pathToPatterns, PaletteIcon],
   ['Новини', routes.pathToNews, ImportContactsIcon],
+  ['Запитання & Відповіді', routes.pathToEmailQuestions, LiveHelpIcon],
```
Comment (TorskyM) [inline src/configs/menu-categories.js:22]: I think `Запитання користувачів` better reflects the purpose of this page

File: src/pages/email-questions/email-question-details/email-question-details.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+  const customerRenderData = useMemo(
+    () =>
+      question
+        ? [
+          { title: 'Дата запитання:', value: getTime(question.date) },
+          { title: 'Статус:', value: emailQuestionStatuses[question.status] },
+          { title: 'Відправник:', value: question.senderName },
+          { title: 'Email:', value: question.email },
+          { title: 'Запитання:', value: question.text }
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-details/email-question-details.js:56]: Add these titles into the config file

File: src/pages/email-questions/email-question-details/email-question-details.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  const adminRenderData = useMemo(
+    () =>
+      question && question.answer
+        ? [
+          { title: 'Дата відповіді:', value: getTime(question.answer.date) },
+          {
+            title: 'Адмін:',
+            value: `${question.answer.admin.firstName} ${question.answer.admin.lastName}`
+          },
+          { title: 'Відповідь:', value: question.answer.text }
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-details/email-question-details.js:71]: Add these titles into the config file

File: src/pages/email-questions/email-question-details/email-question-details.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                  <span>{item.title}</span> {item.value}
+                </Typography>
+              ))}
+
+              {question.status === 'PENDING' && (
+                <TextField
+                  id='filled-full-width'
+                  label='Label'
+                  style={{ margin: 8 }}
+                  placeholder='Відповідь ...'
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-details/email-question-details.js:101]: Move placeholder text into the config file

File: src/pages/email-questions/email-question-details/email-question-details.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                  InputLabelProps={{
+                    shrink: true
+                  }}
+                  value={answerValue}
+                  onChange={({ target: { value } }) => setAnswerValue(value)}
+                  variant='filled'
+                  error={!answerValue && shouldValidate}
+                  helperText={
+                    !answerValue && shouldValidate
+                      ? 'Введіть текст для відповіді'
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-details/email-question-details.js:115]: Please, move this title to config file

File: src/pages/email-questions/email-question-details/email-question-details.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+                variant='outlined'
+                onClickHandler={() => {}}
+              />
+            </Link>
+            {question.status === 'PENDING' && (
+              <SaveButton
+                className={styles.controlButton}
+                id='save'
+                type='submit'
+                title='Відповісти'
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-details/email-question-details.js:137]: Please, move this title to the config file

File: src/pages/email-questions/email-question-filter/email-questions-filter.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import PropTypes from 'prop-types';
+
+import { config } from '../../../configs/index';
+
+const { emailQuestionStatuses } = config;
+
+const EmailQuestionsFilter = ({ filterItems, changeHandler }) => (
+  <ButtonGroup size='small' aria-label='small outlined button group'>
+    <Button
+      id='ALL'
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-filter/email-questions-filter.js:12]: Use lowercase here

File: src/pages/email-questions/email-questions-list.js [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+  const questionClickHandler = (id, status) => {
+    if (status === 'SPAM') {
+      const handler = () => dispatch(closeDialog());
+
+      openSuccessSnackbar(
+        handler,
+        'ПОМИЛКА',
+        EMAIL_QUESTION_SPAM_DETAILS,
+        'Зрозуміло',
```
Comment (TorskyM) [inline src/pages/email-questions/email-questions-list.js:89]: Please, move these titles to the config file

File: src/pages/email-questions/email-questions-list.js [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+  if (loading) {
+    return <LoadingBar />;
+  }
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.tableNav}>
+        <Typography variant='h1' className={styles.title}>
+          Запитання & Відповіді
```
Comment (TorskyM) [inline src/pages/email-questions/email-questions-list.js:162]: Use title from config file here

File: src/pages/email-questions/email-questions-list.js [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+        </div>
+      </div>
+      <div className={styles.tableList}>
+        {questions.length ? (
+          <TableContainerGenerator
+            tableTitles={tableTitles}
+            tableItems={questions}
+          />
+        ) : (
+          <h3 className={styles.emptyList}>Список порожній</h3>
```
Comment (TorskyM) [inline src/pages/email-questions/email-questions-list.js:182]: Move this message into the config files

File: src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+          adminId
+        })
+      );
+      setQuestionsToOperate([]);
+    };
+    openSuccessSnackbar(
+      moveToSpam,
+      EMAIL_QUESTIONS_MOVE_TO_SPAM_MESSAGE,
+      '',
+      'У СПАМ'
```
Comment (TorskyM) [inline src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js:44]: Move this message into the config files

File: src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    const deleteQuestions = () => {
+      dispatch(closeDialog());
+      dispatch(deleteEmailQuestions(questionsToOperate));
+      setQuestionsToOperate([]);
+    };
+    openSuccessSnackbar(
+      deleteQuestions,
+      EMAIL_QUESTIONS_MOVE_TO_DELETE_MESSAGE,
+      '',
+      'Видалити'
```
Comment (TorskyM) [inline src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js:58]: Move this message into the config files

File: src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+      >
+        Перемістити у СПАМ
+      </Button>
+      <Button
+        className={styles.spamBtn}
+        variant='contained'
+        onClick={deleteMany}
+        disabled={!questionsToOperate.length}
+      >
+        Видалити вибрані
```
Comment (TorskyM) [inline src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js:78]: Move this message into the config files

File: src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+  return (
+    <>
+      <Button
+        className={styles.spamBtn}
+        variant='contained'
+        onClick={moveQuestionsToSpam}
+        disabled={!questionsToOperate.length}
+      >
+        Перемістити у СПАМ
```
Comment (TorskyM) [inline src/pages/email-questions/operations-buttons/email-questions-operations-buttons.js:70]: Move this message into the config files

File: src/components/dialog-window/dialog-window.js [hunk 16]
```diff
@@ -42,12 +44,23 @@ const DialogWindow = () => {
...
+            title={CANCEL_TITLE}
+            onClickHandler={handleClose}
+          />
+        )}
+        {!showCancelButton && (
+          <StandardButton
+            variant='contained'
+            title={buttonTitle}
+            onClickHandler={onClickHandler}
+          />
```
Comment (TarasKun) [inline src/components/dialog-window/dialog-window.js:59]: I think it is better to use a ternary operator here

**Review by TarasKun — APPROVED** [review]

File: src/pages/business-pages/business-page-list.js [hunk 17]
```diff
@@ -60,8 +60,10 @@ const BusinessPageList = () => {
         <TableContainerRow
           key={index}
```
Comment (TorskyM) [inline src/pages/business-pages/business-page-list.js:61]: Don't use the index as a key property

File: src/pages/email-questions/email-question-details/email-question-details.js [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+                />
+              )}
+            </div>
+          </div>
+          <div className={styles.controlsBlock}>
+            <Link to={config.routes.pathToEmailQuestions}>
+              <StandardButton
+                className={styles.controlButton}
+                id='back'
+                title='Назад'
```
Comment (TorskyM) [inline src/pages/email-questions/email-question-details/email-question-details.js:153]: Move this title to configs

**Review by TorskyM — APPROVED** [review]

**Pull request merged**